apps/dashboard/src/components/conditions-editor/variable-select.tsx

```diff
@@ -187,7 +187,8 @@ export const VariableSelect = ({
       e.preventDefault();
     } else if (e.key === 'Enter') {
       if (hoveredOptionIndex !== -1) {
-        onSelect(options[hoveredOptionIndex].value ?? '');
+        e.preventDefault();
+        onSelect(filteredOptions[hoveredOptionIndex].value ?? '');
```

Comment:
fix for the `field` component when the values are filtered and user hits the enter